EU4toV2/Source/V2World/Diplomacy/Diplomacy.cpp

```diff
@@ -218,6 +219,35 @@ void V2::Diplomacy::convertRelationsToInfluence(const std::map<std::string, std:
 	}
 }
 
+void V2::Diplomacy::sphereHRE(const EU4::World& sourceWorld, std::map<std::string, std::shared_ptr<Country>>& countries)
+{
+	if (const auto& hreReforms = sourceWorld.getHREReforms();
+		 std::find(hreReforms.begin(), hreReforms.end(), "emperor_reichskrieg") == hreReforms.end())
+	{
+		return;
+	}
+	else
+	{
+		Log(LogLevel::Info) << "\tSphereing HRE";
+
+		const auto& emperorItr = std::find_if(countries.begin(), countries.end(),
+			[](const std::pair<std::string, std::shared_ptr<Country>>& country)
+			{
+				return country.second->isEmperorHRE();
+			});
+		const auto& emperor = emperorItr->second;
```

Comment:
Add logic to handle to HRE

Reply:
find up there may not find the emperor. check if there's something first.